src/ripple/app/ledger/LedgerHistory.cpp

```diff
@@ -158,26 +159,114 @@ log_metadata_difference(Ledger::pointer builtLedger, Ledger::pointer validLedger
     TransactionMetaSet::pointer builtMetaData;
     builtLedger->getTransactionMeta(tx, builtMetaData);
     assert(validMetaData != nullptr || builtMetaData != nullptr);
+
     if (validMetaData != nullptr && builtMetaData != nullptr)
     {
-        WriteLog (lsERROR, LedgerMaster) <<
-            "MISMATCH tx differ in metadata only " << tx <<
-            " : built metadata is " << builtMetaData->getJson(0) <<
-            " : valid metadata is " << validMetaData->getJson(0);
+        auto const& validNodes = validMetaData->getNodes ();
+        auto const& builtNodes = builtMetaData->getNodes ();
+
+        bool result_diff =
```

Comment:
Could make `result_diff` `const` like `index_diff` and `nodes_diff`.  I'd like that.